dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/WorkFlowLineageController.java

```diff
@@ -106,4 +115,36 @@ public Result<Map<String, Object>> queryWorkFlowLineage(@ApiIgnore @RequestAttri
             return error(QUERY_WORKFLOW_LINEAGE_ERROR.getCode(), QUERY_WORKFLOW_LINEAGE_ERROR.getMsg());
         }
     }
+
+    /**
+     * Whether task can be deleted or not, avoiding task depend on other task of process definition delete by accident.
+     *
+     * @param loginUser login user
+     * @param projectCode project codes which taskCode belong
+     * @param processDefinitionCode project code which taskCode belong
+     * @param taskCode task definition code
+     * @return Result of task can be delete or not
+     */
+    @ApiOperation(value = "verifyTaskCanDelete", notes = "VERIFY_TASK_CAN_DELETE")
+    @ApiImplicitParams({
+        @ApiImplicitParam(name = "projectCode", value = "PROCESS_DEFINITION_NAME", required = true, type = "Long"),
+        @ApiImplicitParam(name = "processDefinitionCode", value = "PROCESS_DEFINITION_CODE", required = true, type = "processDefinitionCode"),
+        @ApiImplicitParam(name = "taskCode", value = "TASK_DEFINITION_CODE", required = true, dataType = "Long", example = "123456789"),
+    })
+    @PostMapping(value = "/tasks/verify-delete")
+    @ResponseStatus(HttpStatus.OK)
+    @ApiException(TASK_WITH_DEPENDENT_ERROR)
+    @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
+    public Result verifyTaskCanDelete(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
```

Comment:
Does it need to modify UI?

Reply:
Currently, it should not, this method only needs to call in frontend when we close #10878. When the user selects single in the frontend and try to delete it, have to call `verifyTaskCanDelete` to determine task can be deleted or not. 
![image](https://user-images.githubusercontent.com/15820530/178197131-4e69c8dd-3baf-4403-af3b-c2293db04ee9.png)

Reply:
we can not add this method in frontend now because the dependent tasks can be dependence in the same workflow, and it will raise error when we call this method, and task can never be delete

Reply:
and why we should call this method final in frontend, it is because when user makes some batch change in workflow, like add five tasks and remove one task, which the remove task is used in other workflows. And dolphinschelduler will raise error  when users try to save this workflow(expect behavior) but the bad side is the five new adding tasks will not add to database too, I think is not user friendly for users, so maybe we should verify task can delete firstly